SSA/Projects/InstCombine/ForLean.lean

```diff
@@ -123,9 +123,10 @@ lemma ofInt_ofNat (w n : Nat) :
     BitVec.ofInt w (OfNat.ofNat n) = BitVec.ofNat w n :=
   rfl
 
+lemma ofInt_ofNat' : BitVec.ofInt w (OfNat.ofNat (α := ℤ) x ) = x#w := rfl
```

Comment:
Giving types here should just work:

```suggestion
lemma ofInt_ofNat' {x : Nat} : BitVec.ofInt w (OfNat.ofNat x) = x#w := rfl
```

I am once again suspicious of this, though. Doesn't the `OfNat` simp away?

Reply:
We already have `ofInt_ofNat`, but it does not apply. The problem is here:

```
+++ b/SSA/Projects/InstCombine/AliveHandwrittenLargeExamples.lean
@@ -232,6 +232,8 @@ def alive_simplifyMulDivRem805' (w : Nat) :
   simp
   split_ifs with c
   simp
+  -- TODO: This one does not work: simp [(BitVec.ofInt_ofNat)]
+  simp [(BitVec.ofInt_ofNat')]
```

Reply:
I'm very confused, `n#w` is a macro that expands to `BitVec.ofNat`, so `ofInt_ofNat` and `ofInt_ofNat'` should be the same theorem (as in, syntactically equal statements, post elaboration)

Reply:
The upstream issue tracking this is https://github.com/leanprover/lean4/issues/2867